### PR TITLE
Functions and tests to support more endpoints

### PR DESCRIPTION
**Pull request by Nekroze**

### CODE REVIEW

**Comment by Nekroze** [comment]
seems flakey https://travis-ci.org/JumboInteractiveLimited/fatzebra/builds/466730119

**Comment by Alpineroyale** [comment]
yeah, seems to fail on a different php version each run.

**Comment by Nekroze** [comment]
> yeah, seems to fail on a different php version each run.

I have run 51 times on php 7.0 without failure same on 5.6 flake must be CI specific.

File: src/Gateway.php [hunk 1]
```diff
@@ -358,6 +360,39 @@ public function tokenize($card_holder, $card_number, $expiry_date, $cvv) {
...
+	 * Update a tokenized card
+	 * @param string $token the card token
+	 * @param string $expiry_date the card expiry date (mm/yyyy format)
+	 * @param string $alias alias to associate with the card
+	 * @return \StdObject
+	 */
+    public function update_tokenized_card($token, $expiry_date=null, $alias=null) {
+		if(is_null($token) || (strlen($token) === 0)) throw new InvalidArgumentException('Token is a required field.');
+
+		$payload = null;
```
Comment (Nekroze) [inline src/Gateway.php]: this should be `[]` so the index operations in the following if statements do not fail their index assignment. If the api/client code requires null it should detect an empty array and switch inside do_request so the api logic is contained there

**Comment by Nekroze** [comment]
I am going to try reproduce the CI failures locally again as this [latest run](https://travis-ci.org/JumboInteractiveLimited/fatzebra/builds/467284852?utm_source=github_status&utm_medium=notification) failed about half the versions randomly.

**Comment by Nekroze** [comment]
flake again appears to be CI only, cannot reproduce this myself.

**Pull request merged**